formats/json/commonMain/src/kotlinx/serialization/json/internal/JsonIterator.kt

```diff
@@ -84,6 +85,7 @@ private class JsonIteratorArrayWrapped<T>(
      */
     override fun hasNext(): Boolean {
         if (lexer.peekNextToken() == TC_END_LIST) {
+            sawClosingBrace = true
```

Comment:
Perhaps it would be better to add the `finished` flag to check it at the very beginning of the `hasNext` function? like `if (finished) return false`

The difference is not fundamental, but it may be a little easier to read.

Reply:
Yeah, that makes sense